content/concepts/protocols.md

```diff
@@ -1,227 +1,154 @@
+or connection is called **protocol negotiation**.
 
-The first takes two arguments: a protocol id, and a handler function. If an incoming stream request sends an exact
-match for the protocol id, the handler function will be invoked with the new stream as an argument.
-
-#### Using a match function
-
-The second kind of protocol registration takes three arguments: the protocol id, a protocol match function, and the handler function.
+{{% notice "note" %}}
+Not to be mistaken with [stream multiplexers](/concepts/stream-multiplexing), 
+which are solely responsible for the internal streams on a connection and aren't 
+concerned with underlying protocol being used.
+{{% /notice %}}
 
-When a stream request comes in whose protocol id doesn't have any exact matches, the protocol id will be passed through
-all of the registered match functions. If any returns `true`, the associated handler function will be invoked.
+### Matching protocol IDs and versions
 
-This gives you a lot of flexibility to do your own "fuzzy matching" and define whatever rules for protocol matching
-make sense for your application.
+There are different ways to register a protocol handler. For instance, a match function 
+is a flexible method for protocol registration. A match function
+takes three arguments: the protocol ID, a protocol match function, 
+and the handler function as arguments.
 
 ### Dialing a specific protocol
 
-When dialing a remote peer to open a new stream, the initiating peer sends the protocol id that they'd like to use. The remote peer will use
-the matching logic described above to accept or reject the protocol. If the protocol is rejected, the dialing peer can try again.
+When opening a new stream on a connection, the initiating peer sends the protocol 
+ID it would like to use. The remote peer will use the matching logic described above to 
+accept or reject the protocol. If the protocol is rejected, the dialing peer can try again.
 
-When dialing, you can optionally provide a list of protocol ids instead of a single id. When you provide multiple protocol ids, they will
-each be tried in succession, and the first successful match will be used if at least one of the protocols is supported by the remote peer.
-This can be useful if you support a range of protocol versions, since you can propose the most recent version and fallback to older versions
-if the remote hasn't adopted the latest version yet.
+When opening a new stream, the initiator can optionally provide a list of protocol IDs instead 
+of a single ID. When you provide multiple protocol IDs, they will each is tried in succession, 
+and the first successful match will be used if at least one of the protocols is supported by the 
+remote peer. This can be useful if you support a range of protocol versions since you can 
+propose the most recent version and fallback to older versions if the remote hasn't adopted 
+the latest version yet.
```

Comment:
This seems like an API description. We've already explained that one can continue after the "na" above.